drud/ddev
# [docs] Clean up filenames + update links

## The Problem/Issue/Bug:

The documentation source uses inconsistent separators and a few filenames that could be more terse.

## How this PR Solves The Problem:

This consistently uses all lowercase filenames with hyphens as word separators, updating the navigation and links. It will require a few redirects:

- `users/extend/database_types.md` → `users/extend/database-types.md`
- `users/basics/database_management.md` → `users/basics/database-management.md`
- `users/configuration/config_yaml.md` → `users/configuration/config.md`
- `users/topics/whats_in_ddev_dir.md` → `users/topics/directory-tour.md`
- `users/topics/cms_specific_help.md` → `users/topics/cms-specific-help.md`
- `docs/content/Logo_w_text.svg` → `docs/content/logo.svg`

### Bonus Sub-Issue

> ⚠️ The noteworthy one here is `config_yaml.md` → `config.md`, which I consider foreshadowing.

On many pages, conceptual descriptions of config settings refer to `config.yaml` when they don’t need to. It needlessly forces the mechanics of setting the thing into the explanation of what it does, and implies it must be set via YAML when most settings could be set via CLI. Better to leave out the explicit YAML reference in most cases and link to the reference that prominently clarifies—with examples—how to set the value of a given thing.

**Examples**

>  Open your `.ddev/config.yaml` file and check the `docroot` value, (...)

👍 Keep it! This is an instruction pointing you to a file where you can check and edit the `docroot` setting.

> (...), you can easily add them using `web_extra_daemons` in `.ddev/config.yaml`.

🤔 Could go either way. Might be more efficient to say “add them using the `web_extra_daemons` setting” and link directly to the config page, even though someone will not be setting this via CLI.

> You can choose the Node.js version you want to use in `.ddev/config.yaml` with `nodejs_version`.

👎 Change it: “You can specify the Node.js version you want with the `nodejs_version` setting.” 

Shorter sentence, link to the config page which explains how you can set it **and** further points out that `nvm` is there waiting. No need to be overly-prescriptive with the `.ddev/config.yaml` reference here, which makes this sentence harder to parse if the reader doesn’t already have a working knowledge of `.ddev/config.yaml`. It’s safer to assume they know what a config setting is and link them to a page that thoroughly explores config settings.

## Manual Testing Instructions:

[Examine this branch locally](https://ddev.readthedocs.io/en/latest/developers/testing-docs/#preview-changes) or check out the [automatic preview build](https://ddev--4378.org.readthedocs.build/en/4378/).

## Automated Testing Overview:

n/a

## Related Issue Link(s):

#4188

## Release/Deployment notes:

n/a


<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/4378"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
